yfinance/multi.py

```diff
@@ -29,10 +29,10 @@
 from . import shared
 
 
-def download(tickers, start=None, end=None, actions=False, threads=True, ignore_tz=True, 
+def download(tickers, start=None, end=None, actions=False, threads=True, ignore_tz=True,
              group_by='column', auto_adjust=False, back_adjust=False, repair=False, keepna=False,
              progress=True, period="max", show_errors=True, interval="1d", prepost=False,
-             proxy=None, rounding=False, timeout=10, **kwargs):
+             proxy=None, rounding=False, timeout=10):
```

Comment:
Instead of dropping kwargs, can you add it to `_download_*()` calls below. I think real problem is it's not being passed through.

Reply:
Not sure I understand the *args and **kwargs that was added to functions but never used.
Is it to support deprecated arguments or is there plans to use variable args in the future? Then it can be added when needed without breaking the API.

Reply:
https://github.com/ranaroussi/yfinance/blob/fb5c67b3bddb103f4b53f2e2274355a3c166b3d7/yfinance/base.py#L143

Reply:
I've never seen this in other code (unless they are used) and they were reported as errors by the code inspection.
Keeping them just confuses users of the API and will make extra arguments typos in keyword arguments to pass silently.
I would suggest to keep them out. Do you still want them added in _download_*()?

Reply:
Ah now I think I get what you mean.
The history function supports kwargs to pass debug options.
```
            **kwargs: dict
                debug: bool
                    Optional. If passed as False, will suppress
                    error message printing to console.
                raise_errors: bool
                    Optional. If True, then raise errors as
                    exceptions instead of printing to console.
```

I think it would make more sense to just add them as normal keyword arguments. That way coding editors would be able to do auto complete and detect typos.
Converting them to normal keyword arguments would not break the API.

There are plans for many debug related options then I would add it as a normal keyword argument that expects a dict or a special debug options object.

Reply:
They should stay in some form - `debug_mode` useful for debugging, and I use `raise_errors` in my loops. In fact I just asked a user with issue to set `debug_mode` (#1111), so I need to patch `multi.py` anyway.

It shouldn't confuse users because most don't need to know.

Reply:
saw that there is a undocumented internal "many"-option passed to the history function. Looks like a way to supers raising exceptions and printing errors. So it is to not litter the API with internal options.

I would suggest to have a public method without it and a protected method with the implementation that supports it.

Reply:
Ok that's fair